test/e2e/page-objects/pages/home/homepage.ts

```diff
@@ -408,10 +408,19 @@ class HomePage {
    */
   async checkHasAccountSyncingSyncedAtLeastOnce(): Promise<void> {
     console.log('Check if account syncing has synced at least once');
-    await this.driver.wait(async () => {
-      const uiState = await getCleanAppState(this.driver);
-      return uiState.metamask.hasAccountTreeSyncingSyncedAtLeastOnce === true;
-    }, 30000); // Syncing can take some time so adding a longer timeout to reduce flakes
+    await this.driver.waitUntil(
```

Comment:
We now use `waitUntil` to make sure we keep fetching the state until it's not nullish anymore.